nav2_mppi_controller/src/path_handler.cpp

```diff
@@ -186,6 +186,20 @@ void PathHandler::prunePlan(nav_msgs::msg::Path & plan, const PathIterator end)
   plan.poses.erase(plan.poses.begin(), end);
 }
 
+geometry_msgs::msg::PoseStamped PathHandler::getTransformedGoal()
+{
+  auto goal = global_plan_.poses.back();
+  goal.header.stamp = rclcpp::Time(0);
+  if (goal.header.frame_id.empty()) {
+    throw std::runtime_error("Goal pose has an empty frame_id");
```

Comment:
For the `main` version of this PR, make sure to throw https://github.com/ros-navigation/navigation2/blob/main/nav2_core/include/nav2_core/controller_exceptions.hpp#L37-L42 these exceptions insteaad